### PR TITLE
Token refactoring followup

### PR DESCRIPTION
**Pull request by keyboardDrummer**
### What was changed?
Various simple refactoring to tokens. Best to review the individual commits

### How has this been tested?
No functional changes

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyCore/Resolver/TypeCharacteristicChecker.cs [hunk 1]
```diff
@@ -274,13 +274,13 @@ private static void Check(List<TopLevelDecl> declarations, bool isAnExport, Erro
...
             if (member is Field field) {
-              visitor.VisitType(field.Tok, field.Type, field.IsGhost);
+              visitor.VisitType(field.Origin, field.Type, field.IsGhost);
               if (field is ConstantField { Rhs: { } } cf) {
                 visitor.Visit(cf.Rhs, cf.IsGhost);
               }
             } else if (member is Function function) {
               CheckFormals(function.Ins, function.IsGhost, visitor);
-              visitor.VisitType(function.Result?.Tok ?? function.Tok, function.ResultType, function.IsGhost);
+              visitor.VisitType((function.Result != null ? function.Result.Origin : null) ?? function.Origin, function.ResultType, function.IsGhost);
```
[the remark this answers is not in the document]
Comment (keyboardDrummer) [reply Source/DafnyCore/Resolver/TypeCharacteristicChecker.cs:283]: Yeah, guess so.

File: Source/DafnyCore/Triggers/QuantifierSplitter.cs [hunk 2]
```diff
@@ -112,8 +112,7 @@ protected override void VisitOneExpr(Expression expr) {
         Contract.Assert(quantifier.SplitQuantifier == null);
         if (!splits.ContainsKey(quantifier) && AllowsSplitting(quantifier)) {
           splits[quantifier] = SplitQuantifier(quantifier).Select(e => {
-            e.SetTok(new AutoGeneratedOrigin(e.Tok));
-            e.Origin = e.Origin.MakeAutoGenerated();
+            e.SetOrigin(new AutoGeneratedOrigin(e.Origin));
```
Comment (MikaelMayer) [inline Source/DafnyCore/Triggers/QuantifierSplitter.cs:115]: Interesting that MakeAutoGenerated is no longer necessary.

**Review by MikaelMayer — APPROVED** [review]
Looks good to me.

File: Source/DafnyCore/Rewriters/RefinementTransformer.cs [hunk 3]
```diff
@@ -538,27 +538,27 @@ Method CloneMethod(Method previousMethod, List<AttributedExpression> moreEnsures
           req, reads, mod, ens, decreases, dividedBody, refinementCloner.MergeAttributes(previousMethod.Attributes, moreAttributes), null);
       }
       var body = newBody ?? refinementCloner.CloneBlockStmt(previousMethod.Body);
-      var newRange = currentMethod.Origin.MakeRefined(moduleUnderConstruction);
-      var newName = currentMethod.NameNode.Clone(refinementCloner);
+      var origin = currentMethod.Origin;
```
[the remark this answers is not in the document]
Comment (keyboardDrummer) [reply Source/DafnyCore/Rewriters/RefinementTransformer.cs]: This simplification is local to the changes in RefinementTransformer. It is validated by several testfiles.

File: Source/DafnyCore/Triggers/QuantifierSplitter.cs (hunk 2 above)
Comment (keyboardDrummer) [reply Source/DafnyCore/Triggers/QuantifierSplitter.cs:115]: It was being done twice before. `MakeAutoGenerated` does the same as `new AutoGeneratedOrigin(`

**Review by fabiomadge — APPROVED** [review]
Just reviewed the changes since the last review

**Pull request merged**